optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/report/BenchmarkReport.java

```diff
@@ -353,10 +355,12 @@ private void writeBestScoreSummaryChart() {
                 if (singleBenchmarkResult.hasAllSuccess()) {
                     double[] levelValues = ScoreUtils.extractLevelDoubles(singleBenchmarkResult.getAverageScore());
                     for (int i = 0; i < levelValues.length && i < CHARTED_SCORE_LEVEL_SIZE; i++) {
-                        if (i >= datasetList.size()) {
-                            datasetList.add(new DefaultCategoryDataset());
+                        if (isFinite(levelValues[i])) {
+                            if (i >= datasetList.size()) {
+                                datasetList.add(new DefaultCategoryDataset());
+                            }
+                            datasetList.get(i).addValue(levelValues[i], solverLabel, problemLabel);
                         }
-                        datasetList.get(i).addValue(levelValues[i], solverLabel, problemLabel);
```

Comment:
This code is apparently copy-pasted 3 times in this class. Any chance we could extract it into a separate method?

Reply:
We could probably go even one block up, assuming we extract the `singleBenchmarkResult.get...()` into a lambda.

Reply:
Not sure if it's worth the trouble:
If we do intent to do DRY on this code, it can go a lot further than just extracting these 3 lines, such as the for loop etc.
In any case, separate issue, separate PR. Nice to have, but not a blocker for me.